src/VideoDecoderWorker.cpp

```diff
@@ -265,21 +200,18 @@ int VideoDecoderWorker::Decode()
 	return 0;
 }
 
-void VideoDecoderWorker::onRTP(const RTPIncomingMediaStream* stream, const RTPPacket::shared& packet)
+void VideoDecoderWorker::onMediaFrame(const MediaFrame& frame_)
 {
-	//Put it on the queue
-	packets.Add(packet->Clone());
-}
-
-void VideoDecoderWorker::onEnded(const RTPIncomingMediaStream* stream)
-{
-	//Cancel packets wait
-	packets.Cancel();
-}
+	//Clone it
+	auto frame = std::shared_ptr<MediaFrame>(frame_.Clone());
```

Comment:
do the check about the frame type first on 
```suggestion
	//Check it's a video frame
	if (frame->GetType() != MediaFrame::Type::Video)
	{
		Warning("-VideoDecoderWorker::onMediaFrame(): got %s frame [this: %p]\n", MediaFrame::TypeToString(frame->GetType()), this);
		return;
	}
	//Clone it
	auto cloned = std::shared_ptr<MediaFrame>(frame.Clone());
```